### PR TITLE
fix: required boolean vs expression fix

### PR DESCRIPTION
**Pull request by harmbrugge**
Copy of #3318
https://github.com/molgenis/molgenis-emx2/pull/3318

Created new PR due to unexpected commits

### CODE REVIEW

File: apps/molgenis-components/src/components/forms/FormGroup.vue [hunk 1]
```diff
@@ -3,7 +3,9 @@
     <label v-if="label !== null && label !== undefined" :for="id">
       <b> {{ label }}</b>
     </label>
-    <span v-if="required" class="float-right">(required)</span>
+    <span v-if="required === 'true' || required === true" class="float-right">
```
Comment (mswertz) [inline apps/molgenis-components/src/components/forms/FormGroup.vue:6]: I think you should also cover the case of "TRUE"

**Review by mswertz — APPROVED** [review]
one comment, I think you need to also consider the "TRUE" (I noticed that Excel can push in these values and then it is a string value that nobody understands; we could try to fix this elsewhere but I think a trim/lowercase would be helpful.

**Comment by harmbrugge** [comment]
> one comment, I think you need to also consider the "TRUE" (I noticed that Excel can push in these values and then it is a string value that nobody understands; we could try to fix this elsewhere but I think a trim/lowercase would be helpful.

@mswertz Maybe we should put a toLowerCase on the getter and setter in the backend?

File: apps/molgenis-components/src/components/forms/FormGroup.vue (hunk 1 above)
Comment (chinook25) [reply apps/molgenis-components/src/components/forms/FormGroup.vue:6]: Might be worth it to create a computed or a function for this, in order to not get too much logic in the html

File: apps/molgenis-components/src/components/forms/FormGroup.vue [hunk 2]
```diff
@@ -72,13 +74,25 @@ export default {
       <label for="my-id2">required field form group</label>
       <FormGroup id="my-id2"
         label="my label"
-        required
+        required="'true'"
```
Comment (chinook25) [inline apps/molgenis-components/src/components/forms/FormGroup.vue:77]: use ':required' here, without the colon the inner quotes are part of the string

File: apps/molgenis-components/src/components/forms/FormGroup.vue [hunk 3]
```diff
@@ -72,13 +74,25 @@ export default {
...
           <InputString id="my-id2"></InputString>
         </InputGroup>
         </FormGroup>
     </div>
+
+    <div class="mt-5">
+      <label for="my-id3">required field form group boolean type</label>
+      <FormGroup id="my-id3"
+                 label="my label"
+                 required="true"
```
Comment (chinook25) [inline apps/molgenis-components/src/components/forms/FormGroup.vue:89]: same here, :required. Currently the string `true` is passed, not the boolean

File: apps/molgenis-components/src/components/forms/FormGroup.vue (hunk 2 above)
Comment (chinook25) [reply apps/molgenis-components/src/components/forms/FormGroup.vue:77]: The old version would also still be a good example (just passing `required` is shorthand for `:required="true"`)

File: apps/molgenis-components/src/components/forms/formUtils/formUtils.ts [hunk 4]
```diff
@@ -40,7 +40,7 @@ function getColumnError(
     return undefined;
   }
   if (column.required) {
-    if (column.required === "true") {
+    if (column.required === "true" || column.required === true) {
```
Comment (chinook25) [inline apps/molgenis-components/src/components/forms/formUtils/formUtils.ts:43]: if this also needs to check the for uppercase TRUE, it is worth it to create a "isRequired(column.required)" function

File: apps/schema/src/components/ColumnEditModal.vue [hunk 5]
```diff
@@ -521,10 +521,13 @@ export default {
       this.loading = false;
     },
     setupRequiredSelect() {
```
Comment (chinook25) [inline apps/schema/src/components/ColumnEditModal.vue:523]: Would be nice if the default here is false

**Review by chinook25 — CHANGES_REQUESTED** [review]

**Comment by YpeZ** [comment]
Could we also throw `True` in for good measure? That's Python's affirmative boolean and it may also end up in the _required_ column when using scripts in schema creation

File: apps/molgenis-components/src/components/forms/FormGroup.vue [hunk 6]
```diff
@@ -45,6 +45,9 @@ export default {
     },
   },
   methods: {
+    isRequired(required) {
+      return required === "true" || required === "TRUE" || required === true;
+    },
```
Comment (chinook25) [inline apps/molgenis-components/src/components/forms/FormGroup.vue:50]: load from formUtils, so it works the same everywhere?

File: apps/schema/src/components/ColumnEditModal.vue [hunk 7]
```diff
@@ -265,29 +265,35 @@
 }
 </style>
 
-<script lang="ts">
+<script>
```
Comment (chinook25) [inline apps/schema/src/components/ColumnEditModal.vue:268]: please don't undo the typescripting and removal of unused imports

File: apps/molgenis-components/src/components/forms/formUtils/formUtils.ts [hunk 8]
```diff
@@ -86,6 +86,15 @@ export function isMissingValue(value: any): boolean {
   return value === undefined || value === null || value === "";
 }
 
+export function isRequired(value: any): boolean {
```
Comment (chinook25) [inline apps/molgenis-components/src/components/forms/formUtils/formUtils.ts:89]: Missing unit test

File: apps/molgenis-components/src/components/forms/formUtils/formUtils.ts [hunk 9]
```diff
@@ -86,6 +86,15 @@ export function isMissingValue(value: any): boolean {
...
+  return (
```
Comment (chinook25) [inline apps/molgenis-components/src/components/forms/formUtils/formUtils.ts:90]: Should this function also parse the expressions?

**Review by chinook25 — CHANGES_REQUESTED** [review]

File: apps/molgenis-components/src/components/forms/FormGroup.vue [hunk 10]
```diff
@@ -72,13 +75,25 @@ export default {
...
+        :required="true"
         description="my description">
         <InputGroup>
           <InputString id="my-id2"></InputString>
         </InputGroup>
         </FormGroup>
     </div>
+
+    <div class="mt-5">
+      <label for="my-id3">required field form group boolean type</label>
```
Comment (chinook25) [inline apps/molgenis-components/src/components/forms/FormGroup.vue:87]: They are both boolean, recommend changing the first one to `required="true"`

File: apps/molgenis-components/src/components/forms/formUtils/formUtils.ts [hunk 11]
```diff
@@ -86,6 +86,16 @@ export function isMissingValue(value: any): boolean {
...
 }
 
+export function isRequired(value: string | boolean): boolean {
+  if (typeof value === "string") {
+    if (value.toLowerCase() === "true") return true;
+    if (value.toLowerCase() === "false") return false;
+  } else {
+    return value;
+  }
+  return false;
```
Comment (chinook25) [inline apps/molgenis-components/src/components/forms/formUtils/formUtils.ts:96]: Still, no unit test.

**Review by chinook25 — CHANGES_REQUESTED** [review]
Unit test still missing.

**Pull request merged**